### PR TITLE
repo: add documentation

### PR DESCRIPTION
**Pull request by wilu-nordic**

### CODE REVIEW

File: doc/includes/build_and_run.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+This sample can be found under |sample path| in the |NCS| folder structure.
```
Comment (masz-nordic) [inline doc/includes/build_and_run.txt:2]: NCS?

File: doc/_extensions/table_from_rows.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
```
Comment (masz-nordic) [inline doc/_extensions/table_from_rows.py:3]: License should be BSD-3

File: doc/_static/js/ga-tracker.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+// Copyright (c) 2021 Nordic Semiconductor ASA
+// SPDX-License-Identifier: Apache-2.0
```
Comment (masz-nordic) [inline doc/_static/js/ga-tracker.js:2]: License

File: doc/build_doc_in_docker.sh [hunk 4]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+#
+# Copyright (c) 2022 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
```
Comment (masz-nordic) [inline doc/build_doc_in_docker.sh:5]: License

File: doc/_static/css/nrfxlib.css
Comment (masz-nordic) [inline doc/_static/css/nrfxlib.css:1]: Is this really required?

File: doc/_static/css/tfm.css
Comment (masz-nordic) [inline doc/_static/css/tfm.css:1]: Same

File: doc/_static/css/zephyr.css [hunk 5]
```diff
@@ -0,0 +1,56 @@
+/* Home page grid display */
```
Comment (masz-nordic) [inline doc/_static/css/zephyr.css:1]: Same

File: doc/index.rst [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+##############################################
+
+Nordic Semiconductor announces multi-function `Power Management`_ IC (PMIC) with unique system management features.
+
+:ref:`npmx-zephyr <index>` is a repository with zephyr-like samples for Nordic Semiconductor `npmx library`_.
+
+Zephyr also provides a variety of application samples and demos.
+Documentation for those is available in Zephyr's `Samples and Demos`_ section.
+For very simple samples, see the `Basic Samples`_.
+Those samples are a good starting point for understanding how to put together your own application.
```
Comment (masz-nordic) [inline doc/index.rst:13]: No need to explain what other zephyr examples. Just linking to Zephyr's getting started would be ok.

File: doc/index.rst [hunk 7]
```diff
@@ -0,0 +1,22 @@
+.. _index:
+
+Welcome to npmx zephyr PMIC support repository
+##############################################
+
+Nordic Semiconductor announces multi-function `Power Management`_ IC (PMIC) with unique system management features.
+
+:ref:`npmx-zephyr <index>` is a repository with zephyr-like samples for Nordic Semiconductor `npmx library`_.
```
Comment (masz-nordic) [inline doc/index.rst:8]: Also, driver integration.

File: doc/additional_resources.rst [hunk 8]
```diff
@@ -0,0 +1,11 @@
+.. _additional_resources:
+
+Additional resources
+####################
+
+For additional resources, refer to the following list:
+
+* `Zephyr Getting Started Guide`_
+* `Power Management`_
```
Comment (masz-nordic) [inline doc/additional_resources.rst:9]:
```suggestion
* `Nordic Semiconductor - Power Management ICs`_
```

File: doc/includes/build_and_run.txt [hunk 9]
```diff
@@ -0,0 +1,5 @@
+
+This sample can be found under |sample path| in the `npmx-zephyr repository`_ folder structure.
+
+To build the sample with |VSC|, follow the steps listed on the `How to build an application`_ page in the |nRFVSC| documentation.
+See `Building and programming an application`_ for other building and programming scenarios and `Testing and debugging an application`_ for general information about testing and debugging.
```
Comment (masz-nordic) [inline doc/includes/build_and_run.txt:5]: All those links redirect to NCS. I think vanilla Zephyr docs should be used instead.

File: doc/links.txt [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+.. _`Testing and debugging an application`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/getting_started/testing.html
+.. _`Configuring your application`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/getting_started/modifying.html#configure-application
+.. _`How to connect with PuTTY`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/getting_started/testing.html#how-to-connect-with-putty
+.. _`How to build an application`: https://nrfconnect.github.io/vscode-nrf-connect/get_started/build_app_ncs.html
+.. _`Zephyr Getting Started Guide`: https://docs.zephyrproject.org/latest/develop/getting_started/index.html
+.. _`npmx library`: https://github.com/NordicSemiconductor/npmx
+.. _`nPM1300`: https://www.nordicsemi.com/Products/nPM1300
+.. _`Power Management`: https://www.nordicsemi.com/Products/Power-Management
+.. _`Samples and Demos`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/samples/index.html#samples-and-demos
+.. _`Basic Samples`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/samples/basic/basic.html#basic-sample
```
Comment (masz-nordic) [inline doc/links.txt:16]: Links to [zephyr](https://docs.zephyrproject.org/latest/index.html) would be better

File: doc/links.txt [hunk 11]
```diff
@@ -0,0 +1,16 @@
+.. ### This page lists links used in NCS docs.
+       Links are grouped by source URL if three or more have the same source URL.
+       Otherwise, they are placed in "Source: other".
+       Links not separated by blank line are related to one another.
+
```
Comment (masz-nordic) [inline doc/links.txt:5]:
```suggestion
```

File: doc/index.rst [hunk 12]
```diff
@@ -0,0 +1,17 @@
+.. _index:
+
+Welcome to npmx zephyr PMIC support repository!
```
Comment (masz-nordic) [inline doc/index.rst:3]:
```suggestion
Welcome to `npmx-zephyr` repository!
```

File: doc/index.rst [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+###############################################
+
+:ref:`npmx-zephyr <index>` is a repository with zephyr-like samples for Nordic Semiconductor `npmx library`_ and driver integration with `zephyr`_ environment.
```
Comment (masz-nordic) [inline doc/index.rst:6]:
```suggestion
:ref:`npmx-zephyr <index>` is a repository showcasing the integration of `npmx library`_ into an RTOS environment, using `Zephyr`_ RTOS as an example.
```
Is it ok to use `Zephyr`? Will the link still work?

File: doc/index.rst [hunk 14]
```diff
@@ -0,0 +1,17 @@
...
+
+If you want to see how samples work in zephyr, see `Samples and Demos`_ section.
```
Comment (masz-nordic) [inline doc/index.rst:8]:
```suggestion
If you want to see how samples work in Zephyr, see `Samples and Demos`_ section.
```

File: doc/npmx_zephyr_release_notes.rst
Comment (masz-nordic) [inline doc/npmx_zephyr_release_notes.rst:1]: just `release_notes.rst`

File: license-templates/LICENSE.txt
Comment (masz-nordic) [inline license-templates/LICENSE.txt:1]: Can this be moved to `scripts` instead?

Comment (wilu-nordic) [reply license-templates/LICENSE.txt:1]: Moved

File: doc/npmx_zephyr_release_notes.rst
Comment (wilu-nordic) [reply doc/npmx_zephyr_release_notes.rst:1]: Ok

File: doc/index.rst (hunk 13 above)
Comment (wilu-nordic) [reply doc/index.rst]: `Zephyr` works good

File: doc/sphinx_build_doc_in_docker.sh [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+# The scripts pulls official python docker image, starts container and inside container
+# installs required Python packages (from requirements-doc.txt) and execute sphinx-build tool.
+# HTML documentation is generated to build/html folder
+
+#!/usr/bin/env bash
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: BSD-3-Clause
+#
```
Comment (masz-nordic) [inline doc/sphinx_build_doc_in_docker.sh:19]: Two licenses?

Comment (wilu-nordic) [reply doc/sphinx_build_doc_in_docker.sh]: FIxed

**Review by masz-nordic — APPROVED** [review]

**Comment by mkleszcz-no** [comment]
Will be merged with some changes in:
 - https://github.com/NordicSemiconductor/npmx-zephyr/pull/18

**Pull request closed without merging**